### ISSUE TITLE
Socket test failure, skipping hub connection -and- connect() takes exactly 2 arguments (1 given)

### ISSUE THREAD

**Issue by mikelauff** [FlyingDiver/Indigo-Harmony#14]
Hi. First off, _THANK YOU_ for this Indigo-Harmony plugin that I've been using successfully for over a year. I'm not sure what's changed in my setup, but the plugin stopped working. I am able to control my A/V equipment using the Harmony remote and app, so that's OK. I have verified that XMPP is NOT disabled (and tried disabling it and enabling it again). I've restarted the hub many times. I've checked the hub's IP address in the Harmony iOS app under About > Versions and found 10.0.0.238, which matches what my router says the hub is using, and it matches what the Indigo device (using the plugin) is set to. I've tried downgrading the plugin to versions 7.2.3 and 7.3.0, as well as using the latest 7.3.1 version. 

**Here's what the log shows after starting up v7.3.0:**

> Sep 1, 2020 at 9:29:26 PM
>    Enabling plugin "Harmony Hub 7.3.0"
>    Starting plugin "Harmony Hub 7.3.0" (pid 19537)
>    Harmony Hub Debug               logLevel = 5
>    Started plugin "Harmony Hub 7.3.0"
>    Harmony Hub                     Starting Harmony Hub
>    Harmony Hub Debug               Called deviceStartComm(self, device): Harmony Hub (26069710)
>    Harmony Hub Debug               Harmony Hub: Device Current Version = 1
>    Harmony Hub Debug               Harmony Hub: Device Version is up to date
>    Harmony Hub Debug               Harmony Hub: Starting harmonyHub device (26069710)
>    Harmony Hub Warning             Harmony Hub: Socket test failure, skipping hub connection
>    Harmony Hub Error               Error in plugin execution runConcurrentThread:
> 
> Traceback (most recent call last):
>   File "plugin.py", line 71, in runConcurrentThread
> TypeError: connect() takes exactly 2 arguments (1 given)
> 
>    Harmony Hub Error               plugin runConcurrentThread function returned or failed (will attempt again in 10 seconds)
>    Harmony Hub Error               Error in plugin execution runConcurrentThread:
> 

**And here's what the log shows after starting up v7.3.1:**

> Sep 1, 2020 at 9:59:28 PM
>    Enabling plugin "Harmony Hub 7.3.1"
>    Starting plugin "Harmony Hub 7.3.1" (pid 20021)
>    Harmony Hub Debug               logLevel = 5
>    Started plugin "Harmony Hub 7.3.1"
>    Harmony Hub                     Starting Harmony Hub
>    Harmony Hub Debug               Called deviceStartComm(self, device): Harmony Hub (26069710)
>    Harmony Hub Debug               Harmony Hub: Device Current Version = 1
>    Harmony Hub Debug               Harmony Hub: Device Version is up to date
>    Harmony Hub Debug               Harmony Hub: Starting harmonyHub device (26069710)
>    Harmony Hub Warning             Harmony Hub: Socket test failure, skipping hub connection

**Not much there, but after selecting a control to trigger a TV-watching routine more errors appear...**

> Sep 1, 2020 at 10:00:04 PM
>    WebServer                       request to execute Watch_TV control #13 from xx.xx.xx.xx (IP Address masked in issue submission only)
>    Harmony Hub Debug               Harmony Hub: Start Activity - 23333362
>    Harmony Hub Error               Error in plugin execution ExecuteAction:
> 
> Traceback (most recent call last):
>   File "plugin.py", line 230, in startActivity
>   File "plugin.py", line 246, in doActivity
> AttributeError: 'HubClient' object has no attribute 'client'
> 
> 
> Sep 1, 2020 at 10:00:37 PM
>    Reloading plugin "Harmony Hub 7.3.1"
>    Stopping plugin "Harmony Hub 7.3.1" (pid 20021)
> 
> Sep 1, 2020 at 10:00:53 PM
>    Error                           process (pid 20021) failed to quit after polite request -- forcing it to quit now
>    Stopped plugin "Harmony Hub 7.3.1"
>    Starting plugin "Harmony Hub 7.3.1" (pid 20052)
>    Harmony Hub Debug               logLevel = 5
>    Started plugin "Harmony Hub 7.3.1"
>    Harmony Hub                     Starting Harmony Hub
>    Harmony Hub Debug               Called deviceStartComm(self, device): Harmony Hub (26069710)
>    Harmony Hub Debug               Harmony Hub: Device Current Version = 1
>    Harmony Hub Debug               Harmony Hub: Device Version is up to date
>    Harmony Hub Debug               Harmony Hub: Starting harmonyHub device (26069710)
>    Harmony Hub Warning             Harmony Hub: Socket test failure, skipping hub connection
> 
> Sep 1, 2020 at 10:01:10 PM
>    Harmony Hub Error               Error in plugin execution UiValidate:
> 
> Traceback (most recent call last):
>   File "plugin.py", line 390, in dumpFormattedConfig
> AttributeError: 'HubClient' object has no attribute 'config'

I appreciate any help you're able to provide.

**Comment by FlyingDiver** [comment]
Triple check the IP number of the Hub.  This line means the plugin can't reach it:

    Harmony Hub Warning Harmony Hub: Socket test failure, skipping hub connection

**Comment by mikelauff** [comment]
I have checked many times before reaching out to you. The Hub has been assigned a static IP address of 10.0.0.238. The router and Harmony app both reflect this. I've even disabled the Indigo device and created a new one using the same IP address, but I get the same errors. Anything else you can think of for me to try?

**Comment by mikelauff** [comment]
FYI, if I try one of the **write config to log** options I see this:
![Screen Shot 2020-09-01 at 10 03 43 PM](https://user-images.githubusercontent.com/138063/91926407-328bc080-eca5-11ea-8a3e-4223d4237635.png)

**Comment by FlyingDiver** [comment]
Yeah, that's because the plugin is (internally) disabling the hub because it can't communicate with it.  So there's no config data to write to the log.

**Comment by mikelauff** [comment]
What port is the Hub supposed to be listening on? Perhaps I can use telnet from the Indigo host server Mac to try to telnet to the Hub, just to confirm it is reachable.

**Comment by FlyingDiver** [comment]
Port 5222.  That's what the plugin attempts to create a socket to connect to.

What's really weird is that when I restart the plugin, I get the same failure to connect message.  But I can still control the Hub.  Hmmm.

**Comment by mikelauff** [comment]
So after trying:
`telnet 10.0.0.238 5222`
several times with two different results (either a "host down" or timeout) I decided to shut down Indigo's server, then start it up again. (I had already rebooted my router a few times a day or two ago, and rebooted the Indigo Mac then too.)

When I started the Indigo server again I tried the telnet command and this time it connected. The plugin is happy again, and everything is working. 

I wish I could tell you what I did, but short of stopping the local Indigo server and starting it again (without quitting Indigo) that's all I did. 

Thanks for your suggestions, and fast response! 

I really appreciate the plugin!
